[Fabric] Update to Minecraft 1.20.4

retrixe opened the pull request:
This PR includes commits for 1.19.3-1.21.1 (please use the "Rebase and Merge" option instead of squashing these commits together)

Downloads: https://modrinth.com/mod/torohealth-damage-indicators-updated

Known issues:
- Damage particles don't appear on-screen with shaders enabled in Iris. (Confirmed on 1.20-1.20.4 versions, fixed in 1.20.5+)

retrixe committed:
Port to Minecraft 1.19.3
- Upgrade to Minecraft 1.19.3
- Upgrade to Gradle 7.6
- Upgrade to Fabric Loom 1.0
- Upgrade to Fabric Loader 0.14.11
- Upgrade to Yarn Mappings 1.19.3+build.3
- Upgrade to Fabric API 0.69.1+1.19.3
- Change build.gradle, settings.gradle to fix warnings.
- Change references that were renamed in updated Yarn (shaders, canHit).
- Migrate to JOML, which replaced Minecraft's built-in math classes.

PRRB commented:
Works for me. Thx.

papertek commented:
1.19.2?

retrixe commented:
> 1.19.2?

The current version on CurseForge already works on 1.19.2.

kkellogg378 commented:
1.19.4 only partially works. Makes the in-game GUI green though

brunociria commented:
@kkellogg378 how did you manage to make 1.19.4 work? I tested with the latest fabric version + latest fabric api and It crashes when looking at mobs. I also tested on quilt and I have the same issue.

kkellogg378 commented:
> @kkellogg378 how did you manage to make 1.19.4 work? I tested with the latest fabric version + latest fabric api and It crashes when looking at mobs. I also tested on quilt and I have the same issue.

Unsure. I did run it alongside Sodium and Starlight though, so rendering might have been better. 

Either way, looking at a mob made it laggier and the GUI was all green, so it doesn't work well

retrixe committed:
Update to Minecraft 1.19.4
Includes updated Gradle, Loom, Yarn, Fabric Loader and Fabric API.

retrixe commented:
I'm not sure how the GUI was green and caused fps lag for you, the 1.19.3 version crashed for me with Sodium, Starlight and some other mods like Lithium/Iris/etc, just like the previous user i.e. whenever looking at mobs.

Anyway, I have pushed a 1.19.4 updated version which you can test and see if you have the same issue.

kkellogg378 commented:
![2023-04-03_12 13 54](https://user-images.githubusercontent.com/32850733/229567731-65ec62ad-b9d2-432c-b44e-57c53bddc33f.png)

I usually disable the HUD "showEntity" "showBar" and "showSkin" and set the inWorld "mode" to ALWAYS, and the new 1.19.4 one still changes the item I'm holding to be green (on the bright side, the hotbar is correctly colored now). 

Note: I did test this with just Fabric API and Torohealth enabled, the same thing happens. Also, looking at a hostile enemy makes the held item look red

EDIT: Also no fps issues this time. I think last time my computer was just having a bad day

retrixe committed:
Fix hand rendering issue with in-world health bars

retrixe commented:
I was able to reproduce this issue, I have updated the download JAR with a version that fixes it for me (i.e. the above commit)

kkellogg378 commented:
Works great now! Thank you!

Annijang commented:
> Includes updated Gradle, Loom, Yarn, Fabric Loader and API.
> 
> This PR includes commits for both 1.19.3 and 1.19.4 (preferably use the "Rebase and Merge" option instead of squashing?)
> 
> Downloads (uploaded as ZIP cause GitHub doesn't allow JAR uploads):
> 
> * 1.19.3 version: [torohealth-1.19.3-fabric-1.zip](https://github.com/ToroCraft/ToroHealth/files/10277347/torohealth-1.19.3-fabric-1.zip)
> * 1.19.4 version: [torohealth-1.19.4-fabric-1.zip](https://github.com/ToroCraft/ToroHealth/files/11141909/torohealth-1.19.4-fabric-1.zip)

hi, can you please do 1.19.2forge, 1.19.3forge, 1.19.4forge? Thanks.

retrixe commented:
> hi, can you please do 1.19.2forge, 1.19.3forge, 1.19.4forge? Thanks.

Sorry, I don't have experience with Forge development. I could look into it in the future if I have spare time, but it's very unlikely since I have a lot of my own projects and work, hopefully someone else can step in to update the Forge version.

Annijang commented:
> > hi, can you please do 1.19.2forge, 1.19.3forge, 1.19.4forge? Thanks.
> 
> Sorry, I don't have experience with Forge development. I could look into it in the future if I have spare time, but it's very unlikely since I have a lot of my own projects and work, hopefully someone else can step in to update the Forge version.

Can you make a version of 1.20 when it comes out? Thank you, forget it if you can't forge, it's ok.

HowardZHY commented:
now 1.20.1 when

fijidarko commented:
Any update for 1.20 happening?

retrixe commented:
Minecraft changed rendering a fair bit in 1.20, I have a partial port locally but there's some bugs before I can push it. No ETA

retrixe committed:
Update to Minecraft 1.20.1 (part 1)
- Switch to DrawContext where needed.
- Update Gradle, Loom, FLoader/FAPI, Yarn.

retrixe committed:
Update to Minecraft 1.20.1 (part 2)
Passthrough VertexConsumerProvider where needed.

retrixe commented:
Done, and seems to work without any issues :tada:

Kxsumi commented:
W

HowardZHY commented:
> Done, and seems to work without any issues 🎉

I think you may publish it as ToroHealth Unoffical on Modrinth

mbassie commented:
> 

Bars are working, but can't seem to get the damage particles on screen on 1.20. Still using Iris/Sodium etc.

ZsoltMolnarrr commented:
Hey @retrixe !
Installed the 1.20.1 version alongside Sodium and Iris.
The damage indicator particles don't show at all. Could you please fix it?

MC-Gary003 commented:
Forge? 1.20.1?

ZsoltMolnarrr commented:
1.20.1 Fabric, where the damage particles don’t work fully.
Maybe it’s the same for Forge, haven’t tested it.

MC-Gary003 commented:
But **where's FORGE**

retrixe commented:
Damage particles for me do work with shaders off in Iris/Sodium, but with shaders on in Iris, they don't work, and I'm not sure why.. I don't have any expertise wrt rendering in Minecraft and the code seems ok to me so I'm not sure where to start to get to the bottom of this issue.

> Forge? 1.20.1?

I already made it clear here https://github.com/ToroCraft/ToroHealth/pull/171#issuecomment-1575592333

wsxiya commented:
Why are damage numbers black?

gitguette commented:
> Why are damage numbers black?

same, the bar also doesn't show up.

coolGi69 commented:
Got my own fork for forge 1.20.1 at: https://github.com/coolGi2007/ToroHealth/tree/forge-1.20.1

[torohealth-1.20.1-forge-1.zip](https://github.com/ToroCraft/ToroHealth/files/12908807/torohealth-1.20.1-forge-1.zip)

retrixe committed:
Update to Minecraft 1.20.2
- Fix F3 debug HUD detection.
- Handle new split textures correctly.
- Update Gradle, Loom, FLoader/FAPI, Yarn.

lxly9 commented:
Hiya, using version 1.20.1 fabric and im not getting the in-world bar to show up. This is my config which should work fine (its essentially the same as in 1.19.2) 
![image](https://github.com/ToroCraft/ToroHealth/assets/102386118/c9ae37b5-8445-4b54-978d-fcb3cd6a6ca1)

MrOz59 commented:
same here, in world bars don't appear

qguo-gid commented:
anyone know who to config the bar in the mod to hearts? this is for 1.20.1 version

retrixe commented:
@lxly9 I just tried the 1.20.1 version and health bars are working fine with Sodium+Iris (shaders enabled and disabled) and vanilla configurations
Can you let me know what mods you're running?

@qguo-gid I'm not sure there is such a feature, I haven't added any features to the mod and just updated it as-is

MrOz59 commented:
> @lxly9 I just tried the 1.20.1 version and health bars are working fine with Sodium+Iris (shaders enabled and disabled) and vanilla configurations Can you let me know what mods you're running?
> 
> @qguo-gid I'm not sure there is such a feature, I haven't added any features to the mod and just updated it as-is

I don't know which ones lxly9 is using, but I have a lot of mods, full list here:
https://pastebin.com/gLY2dx5D

retrixe commented:
These are a lot of mods which unfortunately I can't sift through and test myself, I advise trying a binary search (remove half the mods and see if the bug persists) to narrow down any potential conflicts with other mods?

BoardAnims commented:
In 1.20.1 Forge, the damage numbers display black, no matter if the color values are changed in the config files...

TecEash1 commented:
Hi, is there a port to 1.20x/1.21 happening by any chance?

Jszp03 commented:
Can you add mobs to blacklist? 

Cobblemon mobs is incompatible because all cobblemons have different health system, and it already includes healthbars for them so it overlaps

ditt0132 commented:
what about 1.21? someone have a progress? or i can try

RealAprilF00lz commented:
this happened on a different unofficial updated torohealth mod too https://www.curseforge.com/minecraft/mc-mods/torohealth-unofficial

1.20.1-forge-1
when the hud of this mod activates, the name on the boss bar of certain bosses, like l_ender's cataclysm bosses and the luxtructosaurus from alex's caves, will move to another part of the screen, leaving only its shadow on the boss bar. if there is more than one boss bar, this only happens if that boss bar is below every other boss bar

kkellogg378 commented:
Any updates for 1.21.1 fabric?

tangjin0418 commented:
@kkellogg378 
[torohealth-1.21.1.zip](https://github.com/user-attachments/files/17611071/torohealth-1.21.1.zip)
Try this, sorry the community no source code (I deleted it because I originally just made for for personal use)

kkellogg378 commented:
> @kkellogg378 [torohealth-1.21.1.zip](https://github.com/user-attachments/files/17611071/torohealth-1.21.1.zip) Try this.
> 
> Sorry the community, no source code (I deleted it because I originally just made for personal use)

This ended up having some odd graphical issues - the bars wouldn't be anywhere near their respective mobs. 

I did notice there was a fork that works really well though: [healthbars](https://github.com/Fuzss/healthbars)

retrixe commented:
> Can you add mobs to blacklist?
> 
> Cobblemon mobs is incompatible because all cobblemons have different health system, and it already includes healthbars for them so it overlaps

I've been busy with college for the greater part of a year, I'm looking into 1.20.5/6, 1.21.0/1 and 1.21.2/3 updates now

Let me know if you're still interested for me to look into adding this feature

retrixe committed:
Update to Minecraft 1.20.5/1.20.6
- Use FAPI's WorldRenderEvents instead of custom mixin (we already depend on FAPI).
- Adapt to minor API changes.
- Require Java 21 as the baseline.
- Update Gradle, Loom, FLoader/FAPI, Yarn.

retrixe committed:
Update to Minecraft 1.21/1.21.1
- Adapt to minor API changes.
- Update Gradle, Loom, FLoader/FAPI, Yarn.

retrixe committed:
WIP: Update to Minecraft 1.21.2/1.21.3
- Adapt to minor API changes.
- Update Gradle, Loom, FLoader/FAPI, Yarn.

retrixe committed:
WIP: Experimental patches for 1.21.2/1.21.3

retrixe committed:
Fix damage particle colours, bump version

retrixe commented:
> Why are damage numbers black?

Released new versions for 1.20+ that fix this issue. I've updated the mod till 1.21.4, but due to some bugs, the 1.21.2+ release is delayed

I've edited the PR to note all currently known issues, let me know if I'm missing something

retrixe commented:
Update: I've found a fix for damage particles not appearing when using Iris shaders too on 1.20.5+ (unfortunately the fix doesn't work on 1.20-1.20.4) :tada: I'm currently republishing the versions onwards as a result

With this fix this PR is now bug-free

wsxiya commented:
> Update: I've found a fix for damage particles not appearing when using Iris shaders too on 1.20.5+ (unfortunately the fix doesn't work on 1.20-1.20.4) 🎉 I'm currently republishing the versions onwards as a result
> 
> With this fix this PR is now bug-free

Perhaps you could refer to how other mods of the same type display damage numbers when using Iris in version 1.20.1?

Redls07 commented:
> Update: I've found a fix for damage particles not appearing when using Iris shaders too on 1.20.5+ (unfortunately the fix doesn't work on 1.20-1.20.4) 🎉 I'm currently republishing the versions onwards as a result
> 
> With this fix this PR is now bug-free

The other ToroHealth port does not have this bug, but it has also stopped being supported. Maybe it's worth looking at the solution to this problem from this port?

https://github.com/MikhailTapio/ToroHealth-Unofficial
